lib/file.js

```diff
@@ -103,6 +114,166 @@ async function downloadFile(url, filepath, options) {
     }, options);
 }
 
+
+/**
+ * @typedef {Object} DownloadFileOptions
+ *
+ * @property {Number} [timeout=30000] Optional socket timeout (ms)
+ * @property {Object} [headers] Optional override of request headers
+ * @property {Number} [retryMaxDuration=60000] time to retry until throwing an error (ms)
+ * @property {Number} [retryMaxCount] time to retry until throwing an error, overrides retryMaxDuration (ms)
+ * @property {Number} [retryInitialDelay=100] time between retries, used by exponential backoff (ms)
+ * @property {Boolean} [retryEnabled=true] retry on failure enabled
+ * @property {Boolean} [retryAllErrors=false] whether or not to retry on all http error codes or just >=500
+ * @property {Number} [retryBackoff=2] backoff factor for wait time between retries (defaults to 2.0)
+ * @property {Number} [preferredPartSize] Optional custom preferred part size. Might be adjusted depending on the target.
+ * @property {Number} [maxConcurrent] Optional concurrent amount.
+ */
+/**
+ * Download file using multiple simultaneous transfers
+ * Throws the first unrecoverable error if unsuccessful, all others are logged
+ * @param {String|URL} url Location of file to download (e.g. presigned URL) 
+ * @param {String} filepath Path to save file
+ * @param {DownloadFileOptions} options Additional options such as headers, retry features, etc 
+ * @returns {Promise} resolves when download completes
+ */
+async function downloadFileConcurrently(url, filepath, options) {
+    if (options && options.mkdirs) {
+        await ensureDir(path.dirname(path.resolve(filepath)));
+    }
+
+    const downloader = new BlockDownload();
+
+    // extract upload options
+    const downloadOptions = filterObject(
+        options || {},
+        ['headers', ...CONCURRENCY_OPTIONS_FILTER, ...RETRY_OPTIONS_FILTER, 'preferredPartSize']
+    );
+
+    return downloader.downloadFiles({
+        downloadFiles: [
+            {
+                fileUrl: url,
+                filePath: filepath,
+                fileSize: (options && options.fileSize) || undefined
```

Comment:
we need to pass `contentType` through if its passed in so it can be passed to the metadata
```suggestion
                fileSize: (options && options.fileSize) || undefined,
                contentType: (options && options.contentType) || undefined
```

also, note, so we don't forget:
in a follow up change when we integrate this into the pipeline, we need to pass fileSize and contentType here:
https://github.com/adobe/asset-compute-pipeline/blob/master/lib/storage/http.js#L25-L28
```
await http.downloadFile(asset.url, file, {
            retryEnabled: !process.env.ASSET_COMPUTE_DISABLE_RETRIES,
            headers: asset.headers,
            contentType: asset.mimetype,
            fileSize: asset.size
        });
```